### PR TITLE
[onert] Treat shape tensor as flat tensor in shape inference of reshape

### PR DESCRIPTION
**Pull request by periannath**
- This commit treats shape tensor as flat tensor in shape inference of reshape
  - Shape inference for reshape can't accept shape tensor with rank 2 or higher
  - Treat shape tensor as flat tensor to accept rank 2 or higher shape tensor

ONE-DCO-1.0-Signed-off-by: JiHwan Yeo <jihwan.yeo@samsung.com>

---

Related PR : #2170

### CODE REVIEW

**Comment by periannath** [comment]
After offline disscussion with @ragmani, I'll update this PR to support 2D or higher tensors as reshape input. (Ignoring dimension with size 1)

**Comment by periannath** [comment]
@ragmani @hyunsik-yoon I've updated this PR. PTAL

File: runtime/onert/core/src/util/shapeinf/Reshape.cc [hunk 1]
```diff
@@ -91,8 +91,7 @@ void StaticInferer::visit(const ir::operation::Reshape &op)
     const auto *shape_buf = reinterpret_cast<const int32_t *>(shape.data()->base());
     assert(shape_buf);
 
-    assert(shape.shape().rank() == 1);
-    new_shape = convertShape(shape_buf, shape.shape().dim(0), input.shape().num_elements());
+    new_shape = convertShape(shape_buf, shape.shape().num_elements(), input.shape().num_elements());
```
Comment (hyunsik-yoon) [inline runtime/onert/core/src/util/shapeinf/Reshape.cc:94]: Could you also change the name of second param (`const int32_t rank`) of `convertShape()` to align with the real meaning?

Comment (periannath) [reply runtime/onert/core/src/util/shapeinf/Reshape.cc:94]: @hyunsik-yoon Thanks. I've updated.

**Pull request merged**